gateway/transport/server.go

```diff
@@ -167,6 +184,20 @@ func (s *Server) Connect(stream pb.Transport_ConnectServer) (err error) {
 		ParamsData: map[string]any{},
 	}
 
+	// Verifying if the feature is enabled
+	currentAccessMode := GetAccessModesFromConnect(clientVerb[0], clientOrigin[0])
+
+	AccessModes := map[string]string{
+		"exec":     gwctx.Connection.AccessModeExec,
+		"connect":  gwctx.Connection.AccessModeConnect,
+		"runbooks": gwctx.Connection.AccessModeRunbooks}
+
+	if AccessModes[currentAccessMode] == "disabled" {
+		return status.Error(codes.FailedPrecondition,
+			fmt.Sprintf("the %v access mode connection has the %v feature disabled", gwctx.Connection.Name, currentAccessMode))
```

Comment:
The error message seems a bit confuse. Let's keep the same error from the proxy manager:

```go
fmt.Sprintf("the %s connection has the access mode %v feature disabled", gwctx.Connection.Name, currentAccessMode))
```